src/core/api.proto

```diff
@@ -132,6 +169,15 @@ message InferRequestHeader
     //@@       highest probabilities will be returned.
     //@@
     Class cls = 3;
+
+    //@@    .. cpp:var:: InferSharedMemory shared_memory
+    //@@
+    //@@       It is the location in shared memory that the result tensor data
+    //@@       for this output will be written. Using shared memory is optional
+    //@@       but if this message is used, all fields are required. The memory
+    //@@       must be continuous. For non-contiguous, use Dynamic Batching.
```

Comment:
It is too confusing to try to explain contiguous vs non-contiguous here. So drop everything after "... all fields are required".